active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+         /// <summary>
+         /// Interface for Controls that participate in input validation.
+         /// <summary/>
+          [webhosthidden]
+         interface IInputValidationControl 
+         {
+             /// <summary>
+             /// Collection of errors to display based on the InputValidationKind property.
+            /// </summary>
+            Windows.Foundation.Collections.IObservableVector<Windows.UI.Xaml.Controls.InputValidationError> ValidationErrors{ get; };
+            /// <summary>
+            ///  Extra information about the Source of the validation. 
+            ///  Note: This is automatically set by the XAML markup compiler. See summary above. 
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationContext ValidationContext;
+            /// <summary>
+            ///  DataTemplate that expresses how the errors are displayed.
+            /// </summary>
+            Windows.UI.Xaml.DataTemplate ErrorTemplate;
+            /// <summary>
+            ///  Determines how the control should visualize validation errors.
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationKind InputValidationKind;
+            /// <summary>
+            /// Command associated with the IInputValidationControl. See InputValidationCommand for more
+            /// information.
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationCommand ValidationCommand;
+         };
+   
+## InputValidationCommand
+The command allows for easy use of input validation without the need for a tight dependency on x:Bind.  This enables developers who aren't striclty using MVVM to use input validation (i.e ReactiveUI).  The InputValidationCommand is what allows for the correct-by-design approach where controls initially validate on focus lost, but then will query for validation on property changed once in an invalid state.  This wouldn't be for free, but the amount of code here is pretty trivial. We could provide some attribute or pattern that tells the markup compiler how to generate that code automatically.  
+
+The command can shared between different IInputValidationControls, as would most likely be the case in the scenario of a forms control.  A custom command can be made that listens for changes to all errors and knows when a form has been completed and ready for submition. The control will query the command as to whether they should perform validation.
+
+Developers can override any of this behavior by creating their own InputValidationCommand.
+ 
+ 
+    /// <summary>
+    /// Command that controls how an IInputValidationControl does validation.
+    /// </summary>
+    [webhosthidden]
+    unsealed runtimeclass InputValidationCommand
+    {
+        [method_name("CreateInstance")] InputValidationCommand();
```

Comment:
This formatting looks weird. Should `[method_name("CreateInstance")]` be on its own line, as an attribute on the method?

Reply:
I don't think this attribute should be needed.